packages/aws-cdk/lib/api/deployments.ts

```diff
@@ -417,6 +493,120 @@ export class Deployments {
     });
   }
 
+  public async rollbackStack(options: RollbackStackOptions): Promise<RollbackStackResult> {
+    const {
+      stackSdk,
+      resolvedEnvironment: _,
+      cloudFormationRoleArn,
+      envResources,
+    } = await this.prepareSdkFor(options.stack, options.roleArn, Mode.ForWriting);
+
+    if (options.validateBootstrapStackVersion ?? true) {
+      // Do a verification of the bootstrap stack version
+      await this.validateBootstrapStackVersion(
+        options.stack.stackName,
+        BOOTSTRAP_STACK_VERSION_FOR_ROLLBACK,
+        options.stack.bootstrapStackVersionSsmParameter,
+        envResources);
+    }
+
+    const cfn = stackSdk.cloudFormation();
+    const deployName = options.stack.stackName;
+
+    while (true) {
+      let cloudFormationStack = await CloudFormationStack.lookup(cfn, deployName);
+
+      // TODO: currently, we only roll back UPDATE_FAILED and CREATE_FAILED stacks. Conceivably, we could also
+      // handle UPDATE_ROLLBACK_FAILED by offering some option like "force rollback".
+      switch (cloudFormationStack.stackStatus.rollbackChoice) {
+        case RollbackChoice.NONE:
+          warning(`Stack ${deployName} does not need a rollback: ${cloudFormationStack.stackStatus}`);
+          return { notInRollbackableState: true };
+
+        case RollbackChoice.START_ROLLBACK:
+          debug(`Initiating rollback of stack ${deployName}`);
+          await cfn.rollbackStack({
+            StackName: deployName,
+            RoleARN: cloudFormationRoleArn,
+            ClientRequestToken: randomUUID(),
```

Comment:
We might not, but we should.

The token is there so that when the SDK retries because of network trouble, a command that has successfully been received by the server (but its success response hasn't made it to the client yet) will not be performed again (and in this particular case produce an error response, even though the rollback was started successfully)